controllers/helpdesk.js

```diff
@@ -1,44 +1,67 @@
 const express = require('express');
+
 const router = express.Router();
-const api = require('../api');
 const logger = require('winston');
+const UAParser = require('ua-parser-js');
+const api = require('../api');
 
 // secure routes
 router.use(require('../helpers/authentication').authChecker);
 
-router.post('/', function (req, res, next) {
-    if (!req.body.subject && req.body.target) {
-        if(req.body.target === "HPI"){ // Contact Admin
-            // title? Y: Feedback N: Problem
-            req.body.subject = req.body.type + ((req.body.title) ? `: ${req.body.title}` : '');
-        }
-        req.body.type = `contact${req.body.target}`;
-    }
+router.post('/', (req, res, next) => {
+	if (!req.body.subject && req.body.target) {
+		if (req.body.target === 'HPI') { // Contact Admin
+			// title? Y: Feedback N: Problem
+			req.body.subject = req.body.type + ((req.body.title) ? `: ${req.body.title}` : '');
+		}
+	}
+	req.body.type = `contact${req.body.target}`;
+
+	// read User-Agent
+	const parser = new UAParser();
+	parser.setUA(req.headers['user-agent']);
+	const result = parser.getResult();
```

Comment:
you could try/catch that and if it fails just take care here that result.browser and result.os is defined

Reply:
result is always defined be default by UAParaser.
Calling result.os.version without setting an user-agent first will return undefined but not producing an error